pkgs/models/model.nix

```diff
@@ -0,0 +1,85 @@
+{ stdenvNoCC
+, fetchurl
+, makeWrapper
+
+, sticker
+
+# Word and tag embeddings as an attrset, containing the attributes:
+#
+# - embeds: Embedding file fixed-output derivation.
+# - filename: Embedding file name in the model's sticker.conf
+, wordEmbeds
+, tagEmbeds ? null
+
+# Short name of the model. E.g.: de-pos-ud.
+, modelName
+
+# Version of the model, typically a date. E.g.: 20190821
+, version
+
+, sha256
+}:
+
+rec {
+  model = stdenvNoCC.mkDerivation rec {
+    inherit version;
+
+    pname = "sticker-model-${modelName}";
+
+    src = fetchurl {
+      inherit sha256;
+
+      url = let
+        fullName = "${modelName}-${version}";
+        in "https://github.com/stickeritis/sticker-models/releases/download/${fullName}/${fullName}.tar.gz";
+    };
+
+    preConfigure = ''
+      substituteInPlace sticker.conf \
+        --replace "${wordEmbeds.filename}" \
+      "${wordEmbeds.embeds}"
+    '' + stdenvNoCC.lib.optionalString (tagEmbeds != null) ''
+      substituteInPlace sticker.conf \
+        --replace "${tagEmbeds.filename}" \
+      "${tagEmbeds.embeds}"
+    '';
+
+    installPhase = ''
+      mkdir -p $out/share/sticker/models/${modelName}
+      install -m 0644 *.conf *.graph *.labels *.shapes epoch-* $out/share/sticker/models/${modelName}
+    '';
+
+    meta = with stdenvNoCC.lib; {
+      homepage = https://github.com/danieldk/sticker/;
+      description = "Sticker ${modelName} model";
+      license = licenses.unfreeRedistributable;
+      maintainers = with maintainers; [ danieldk ];
+      platforms = platforms.unix;
+    };
+  };
+
+  wrapper = stdenvNoCC.mkDerivation rec {
```

Comment:
Does the part below combine the model parameters, embeddings etc. with the binaries?

Reply:
Realizing this derivation crates a package containing the `sticker-tag-<modelname>` and `sticker-tag-<modelname>` wrapper scripts. These wrapper scripts (respectively) call `sticker-tag` and `sticker-server` from the `sticker` package, with the model from the corresponding `model` package. The resulting scripts look like this:

```
% cat $(nix-build -A models.de-pos-ud.wrapper)/bin/sticker-tag-de-pos-ud
#! /nix/store/l6h4ya0wzb4b8mr0y58k2gh2nhfql4sn-bash-4.4-p23/bin/bash -e
exec "/nix/store/kr6h6ggdifsdwb0ncix13lg22kwr1xrc-rust_sticker-utils-0.6.1/bin/sticker-tag"  /nix/store/rkqydriai1d33d9pdwr41v509fpcwclf-sticker-model-de-pos-ud-20190821/share/sticker/models/de-pos-ud/sticker.conf "${extraFlagsArray[@]}" "$@"
```